app/Services/Manager.php

```diff
@@ -92,6 +92,77 @@ public function exportCSV($model, $hasReportback = false)
     }
 
     /**
+     * Build leaderboard data for a given Competition
+     *
+     * @param Competition $competition
+     * @param User $users
+     * @return array $leaderboard
+     */
+    public function getLeaderboard($competition, $users)
+    {
+        $rows = [];
+
+        // Get all users in bulk
+        $parameters['ids'] = implode(',', $users->pluck('id')->all());
+        $users = $this->northstar->getAllUsers($parameters);
+        $users = collect($users)->keyBy('id')->all();
+
+        foreach ($users as $user) {
+            // For each user get the reportback details
+            $reportback = $this->getUserActivity($user->id, $competition);
```

Comment:
We need to fix up the NS/Phoenix route for the RB retrieval to allow doing it in bulk based on user ID. Chatted about it w/ @deadlybutter for a bit. We could fix up the signups endpoint or add passing users to the `/reportbacks` endpoint since all these user's have already signed up for the campaign and all we really need are their reportbacks.